docs/src/controller.md

```diff
@@ -0,0 +1,120 @@
+
+Kubernetes uses the
+[Controller pattern](https://kubernetes.io/docs/concepts/architecture/controller/)
+to align the current cluster state with the desired one.
+
+Stateful applications are usually managed with the
+[`StatefulSet`](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/)
+controller, which allows to create and align a set of Pods created with the
+same specification and maintaining a sticky identity.
+
+CloudNativePG, instead of relying on the `StatefulSet` controller to manage
+the PostgreSQL instances, implements its own custom controller. This design
+choice, while bringing more complexity in the implementation, provided the
+operator with more flexibility on how we manage the cluster while being
+transparent on the topology of PostgreSQL clusters.
+
+PostgreSQL's operators can be certainly designed around the `StatefulSet`
+behavior (and many are indeed) but CloudNativePG avoid this tight
+coupling. This may surprise new users.
+
+Like many choices in the design realm, different ones lead to different
+compromises. The next sections discuss a few of the points where we believe
+this design choice has made the implementation of CloudNativePG
+more reliable and easy to understand.
+
+## Primary Instances versus Replicas
+
+The `StatefulSet` controller is designed to create a set of Pods
+from just one template. Given that we use one `Pod` per PostgreSQL instance,
+we have two kinds of Pods:
+
+- one for the primary instance, and
+- the other ones for replicas.
+
+While the operator is using the same spec for every Pod and the role
+is managed using labels, the rolling deployment mechanism depends on the
+operation that triggered it.
+
+There are operations that should be applied on the replicas first
+and then on the primary, but only after an updated replica is promoted
+as the new primary.
+This happens for example when you want to apply a different image version,
+or when you lower configuration parameters like `max_connections` (which are
+[treated in a special way by PostgreSQL when hot standby replicas are around](https://www.postgresql.org/docs/current/hot-standby.html)).
+
+While doing that, CloudNativePG takes into consideration the role of
+the PostgreSQL instance - and not just its serial number.
+
+Sometimes the operator needs to follow the opposite process: work on the
+primary first, and then on the replicas. This happens, for example, when you
+raise `max_connections`. In that case CloudNativePG will:
+
+- apply the new setting to the primary instance
+- restart it
+- apply the new setting on the replicas
+
+The `StatefulSet` controller, being application-independent, can't
+incorporate this behavior, which is specific to PostgreSQL's native
+replication technology.
+
+## Coherence of PVCs
+
+Sometimes the same PostgreSQL instance works on multiple PVCs: this happens
+when the WAL storage is separated from `PGDATA`, and in the future when CloudNativePG
+will support tablespaces.
```

Comment:
I would avoid citing the tablespaces because it is not needed to validate the point we are trying to make
```suggestion
```